src/main/java/org/broadinstitute/consent/http/authentication/AuthorizationHelper.java

```diff
@@ -0,0 +1,104 @@
+package org.broadinstitute.consent.http.authentication;
+
+import com.google.common.cache.Cache;
+import com.google.inject.Inject;
+import jakarta.ws.rs.NotFoundException;
+import jakarta.ws.rs.WebApplicationException;
+import java.util.Map;
+import org.broadinstitute.consent.http.filters.ClaimsCache;
+import org.broadinstitute.consent.http.models.AuthUser;
+import org.broadinstitute.consent.http.models.User;
+import org.broadinstitute.consent.http.models.sam.UserStatus;
+import org.broadinstitute.consent.http.models.sam.UserStatusInfo;
+import org.broadinstitute.consent.http.service.UserService;
+import org.broadinstitute.consent.http.service.sam.SamService;
+import org.broadinstitute.consent.http.util.ConsentLogger;
+
+/**
+ * Helper class for authorization and authentication. This class contains methods to build AuthUser
+ * objects from request headers, retrieve user status information from Sam, and check user roles.
+ */
+public class AuthorizationHelper implements ConsentLogger {
+
+  protected final ClaimsCache claimsCache;
+  protected final SamService samService;
+  protected final UserService userService;
+
+  @Inject
+  public AuthorizationHelper(SamService samService,
+      UserService userService) {
+    this.claimsCache = ClaimsCache.getInstance();
+    this.samService = samService;
+    this.userService = userService;
+  }
+
+  protected Cache<String, Map<String, String>> getCache() {
+    return claimsCache.cache;
+  }
+
+  protected AuthUser buildAuthUserFromHeaders(Map<String, String> headers) {
+    String aud = headers.get(ClaimsCache.OAUTH2_CLAIM_aud);
+    String token = headers.get(ClaimsCache.OAUTH2_CLAIM_access_token);
+    String email = headers.get(ClaimsCache.OAUTH2_CLAIM_email);
+    String name = headers.get(ClaimsCache.OAUTH2_CLAIM_name);
```

Comment:
Do we ever really use this value?  I'm wondering if we should get rid of `name` and instead use our User object's displayName property.

Reply:
We do use this. When a user is first signing in, we don't have a display name. We use this to do that initial population of the user's display name. Subsequently, users can update this in their profile.